### PR TITLE
STYLE: fix pylint style checks for inherit-non-class.

### LINKED ISSUE #48855: STYLE fix pylint issues
In https://github.com/pandas-dev/pandas/pull/48759/files we're introduced `pylint`, but have turned off its warnings as there's a _lot_ of them:

https://github.com/pandas-dev/pandas/blob/d719840e5a2639babab3b4646b11a111547c518f/pyproject.toml#L36-L67

Task here is:

1. pick one of the above warnings / errors, and remove it from `pyproject.toml`
2. run `pre-commit run pylint --hook-stage manual --all-files`. If this already passes, skip to step 4.
3. fixup any warnings that result
4. if `pre-commit run pylint --hook-stage manual --all-files` passes, then stage and commit your changes and open a pull request

If you believe a warning is a false-positive, then it's OK to ignore it in-line, e.g. 
```python
df.rename(id, mapper=id)  # pylint: disable=redundant-keyword-arg
```

Please comment here which pylint warning you'll work on before starting so we don't duplicate work. No need to ask for permission to work on this, and no need to comment "take" as multiple people can work on this concurrently

---

NOTE: please do _not_ write "closes #48855" in your PR

---

checks we probably can't turn off:
- too-many-function-args
- unsubscriptable-object
- unsupported-assignment-operation
- no-name-in-module
- no-member
- import-error
- not-an-iterable
- invalid-unary-operator-type
- invalid-name
- wrong-import-order (and anything else related with imports)
- unexpected-keyword-argument
- use-implicit-booleaness-not-len
- use-implicit-booleaness-not-comparison
- comparison-with-itself
- no-else-*
- line-too-long
- overridden-final-method
- singleton-comparison
- unsupported-membership-test
- pointless-statement
- broad-except
- undefined-variable

If you're not sure whether a check should fit into the above list, please do ask

### PR DESCRIPTION
**Pull request by kostyafarber**
- [ ] xref #48855 - one of many style checks, still needs work.
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Added an ignore statement in the `base.py`. Looks like we need to inherit a non-class in this instance.

### CODE REVIEW

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -93,7 +93,9 @@
 
 if TYPE_CHECKING:
 
-    class ExtensionArraySupportsAnyAll("ExtensionArray"):
+    class ExtensionArraySupportsAnyAll(
+        "ExtensionArray"
+    ):  # pylint: disable=inherit-non-class
```
Comment (MarcoGorelli) [inline pandas/core/arrays/base.py:98]: does it work to do
```
class ExtensionArraySupportsAnyAll(ExtensionArray)
```
?

Comment (kostyafarber) [reply pandas/core/arrays/base.py:98]: I'm getting flake8 errors for `pandas/core/arrays/base.py:96:40: F821 undefined name 'ExtensionArray'`.

Comment (kostyafarber) [reply pandas/core/arrays/base.py:98]: This might be a dumb question but is it because `ExtensionArray` is defined below `ExtensionArraySupportsAnyAll`?

Comment (MarcoGorelli) [reply pandas/core/arrays/base.py:98]: Thanks for checking, you're right

Perhaps we can move the definition of `ExtensionArraySupportsAnyAll` to right after the definition of `ExtensionArraySupportsAnyAll`, i.e.

So
```
class ExtensionArray:
    ...

class ExtensionArraySupportsAnyAll(ExtensionArray):
    def any(self, *, skipna: bool = True) -> bool:
        pass

    def all(self, *, skipna: bool = True) -> bool:
        pass
```
?

I'll check again tomorrow if that's fine

Comment (kostyafarber) [reply pandas/core/arrays/base.py:98]: Of course. I'll move it and run the pre-commit and see how it goes.

Comment (kostyafarber) [reply pandas/core/arrays/base.py]: Have moved it below `ExtensionArray`. Passed all pre-commits. Just waiting on CI

File: pandas/core/arrays/base.py [hunk 2]
```diff
@@ -91,21 +91,6 @@
...
-        def any(self, *, skipna: bool = True) -> bool:
-            pass
-
-        def all(self, *, skipna: bool = True) -> bool:
-            pass
-
-    from pandas._typing import (
-        NumpySorter,
-        NumpyValueArrayLike,
-    )
```
Comment (MarcoGorelli) [inline pandas/core/arrays/base.py:106]: sorry, I meant to only move `class ExtensionArraySupportsAnyAll`

So here you'd still have
```
if TYPE_CHECKING:
    from pandas._typing import (
        NumpySorter,
        NumpyValueArrayLike,
    )
```

and below `class ExtensionArray` you'd have `class ExtensionArraySupportsAnyAll`

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/core/arrays/base.py]: cool, seems fine, thanks

File: pandas/core/arrays/base.py (hunk 2 above)
Comment (kostyafarber) [reply pandas/core/arrays/base.py:106]: Ah okay sure thing I'll move that now.

File: pandas/core/arrays/base.py [hunk 3]
```diff
@@ -532,6 +517,7 @@ def dtype(self) -> ExtensionDtype:
 
     @property
     def shape(self) -> Shape:
+
```
Comment (MarcoGorelli) [inline pandas/core/arrays/base.py:520]: also, please check that you don't accidentally introduce extra blank lines

Comment (kostyafarber) [reply pandas/core/arrays/base.py:527]: Sorry about that will change.

**Review by MarcoGorelli — APPROVED** [review]
looks good to me pending green

**Comment by kostyafarber** [comment]
Hi, @MarcoGorelli this might sound like a silly request, but is it possible to add the `hacktoberfest` label to this PR?
Many thanks 😄

**Pull request merged**